core/src/main/java/io/parsingdata/metal/data/ParseGraph.java

```diff
@@ -140,6 +150,7 @@ public boolean equals(final Object obj) {
             && Objects.equals(head, ((ParseGraph)obj).head)
             && Objects.equals(tail, ((ParseGraph)obj).tail)
             && Objects.equals(branched, ((ParseGraph)obj).branched)
+            && Objects.equals(scopeDepth, ((ParseGraph)obj).scopeDepth)
```

Comment:
The `scopeDepth` is essentially _derivable_ right? Maybe it should be considered cached then as well just like `size`?

Reply:
With normal usage, it should be derivable. However, it is possible to create a token that alternates its scope delimiter. This makes it possible to have two identitcal ParseGraphs with different scope depths. This is only theoretical and I do not know why you would create a token like that, but for robustness it is better to keep this here. Other minor reason to keep it here is so we can test equality for the ParseGraph with the AutoEqualityTest.